### PR TITLE
Fix some build errors with `disable_3d=yes`

### PR DESCRIPTION
**Pull request by AThousandShips**
* Some tests were incorrectly included
* SCU builds with animation
* Animation `switch`

I'm unsure if this is the best way with the scu, but it solves the issue, could also wrap the code in `#ifndef` but picked this way for right now, can switch if desired

I'm not sure why the linking fails with the SCU builds *if* the file is along with the others, so worth investigating separately

Edit: Swapped to the plain `#ifndef` version, felt more guaranteed

~The issues with the SCU is link errors, which *might* be MSVC specific, haven't confirmed yet~
Not MSVC specific, happens on mingw as well

### CODE REVIEW

File: scene/animation/animation_mixer.cpp
Comment (AThousandShips) [inline scene/animation/animation_mixer.cpp:1]: This parts errors regardless, a `switch` with only a `default`

File: scene/animation/animation_mixer.cpp [hunk 1]
```diff
@@ -954,15 +954,15 @@ Variant AnimationMixer::post_process_key_value(const Ref<Animation> &p_anim, int
...
 				const Skeleton3D *skel = Object::cast_to<Skeleton3D>(p_object);
 				return Vector3(p_value) * skel->get_motion_scale();
 			}
 			return p_value;
-		} break;
 #endif // _3D_DISABLED
+		} break;
 		default: {
 		} break;
 	}
```
Comment (aaronfranke) [inline scene/animation/animation_mixer.cpp:968]: Considering we are using only one of the track type values, this function might make more sense if we just rewrite it to not use a switch statement.

```cpp
Variant AnimationMixer::_post_process_key_value(const Ref<Animation> &p_anim, int p_track, Variant p_value, const Object *p_object, int p_object_idx) {
#ifndef _3D_DISABLED
	if (p_anim->track_get_type(p_track) == Animation::TYPE_POSITION_3D && p_object_idx >= 0) {
		const Skeleton3D *skel = Object::cast_to<Skeleton3D>(p_object);
		return Vector3(p_value) * skel->get_motion_scale();
	}
#endif // _3D_DISABLED
	return p_value;
}
```

**Review by aaronfranke — COMMENTED** [review]
See also #79185

**Review by aaronfranke — APPROVED** [review]
Looks good to me.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by AThousandShips** [comment]
Thank you!

**Comment by YuriSizov** [comment]
Cherry-picked for 4.2.2.

**Pull request merged**